### PR TITLE
Sage Intacct Redesign : Export Log

### PR DESCRIPTION
**Pull request by anishfyle**

### CODE REVIEW

File: src/app/core/services/si/export-log/export-log.service.ts [hunk 1]
```diff
@@ -25,46 +26,69 @@ export class ExportLogService {
...
-    params.tasklog__status = state;
+    params.state = state;
+
     if (selectedDateFilter) {
       const startDate = selectedDateFilter.startDate.toLocaleDateString().split('/');
       const endDate = selectedDateFilter.endDate.toLocaleDateString().split('/');
-      params.start_date = `${startDate[2]}-${startDate[1]}-${startDate[0]}T00:00:00`;
-      params.end_date = `${endDate[2]}-${endDate[1]}-${endDate[0]}T23:59:59`;
+      params.exported_at__gte = `${startDate[2]}-${startDate[1]}-${startDate[0]}T00:00:00`;
+      params.exported_at__lte = `${endDate[2]}-${endDate[1]}-${endDate[0]}T23:59:59`;
```
Comment (ashwin1111) [inline src/app/core/services/si/export-log/export-log.service.ts:40]: this should be start_date and end_date only

File: src/app/core/services/si/export-log/export-log.service.ts [hunk 2]
```diff
@@ -25,46 +26,69 @@ export class ExportLogService {
...
+
+  getExpensesGroupById(expenseGroupId: number): Observable<ExpenseGroup> {
+    const workspaceId = this.workspaceService.getWorkspaceId();
+    return this.apiService.get(`/workspaces/${workspaceId}/fyle/expense_groups/${expenseGroupId}/`, {});
+  }
+
+  syncExpenseGroups() {
+    const workspaceId = this.workspaceService.getWorkspaceId();
+    return this.apiService.post(`/workspaces/${workspaceId}/fyle/expense_groups/trigger/`, {});
+  }
```
Comment (ashwin1111) [inline src/app/core/services/si/export-log/export-log.service.ts:83]: these are not needed

File: src/app/core/services/si/export-log/export-log.service.ts [hunk 3]
```diff
@@ -25,46 +26,69 @@ export class ExportLogService {
...
+
+  getExpenseGroupSettings(): Observable<ExpenseGroupSetting> {
+    const workspaceId = this.workspaceService.getWorkspaceId();
+    return this.apiService.get(`/workspaces/${workspaceId}/fyle/expense_group_settings/`, {});
+  }
+
+  createExpenseGroupsSettings(expenseGroupSettingsPayload: ExpenseGroupSetting): Observable<ExpenseGroupSetting> {
```
Comment (ashwin1111) [inline src/app/core/services/si/export-log/export-log.service.ts:90]: not needed

File: src/app/core/services/si/export-log/export-log.service.ts (hunk 2 above)
Comment (ashwin1111) [reply src/app/core/services/si/export-log/export-log.service.ts:83]: 70-83

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 4]
```diff
@@ -7,9 +18,210 @@ import { Component, OnInit } from '@angular/core';
...
+      startDate: new Date(new Date().getFullYear(), new Date().getMonth(), new Date().getDate() - new Date().getDay()),
+      endDate: new Date()
+    },
+    {
+      dateRange: 'Today',
+      startDate: new Date(),
+      endDate: new Date()
+    },
+    {
+      dateRange: new Date().toLocaleDateString(),
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts:48]: why this option?

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 5]
```diff
@@ -7,9 +18,210 @@ import { Component, OnInit } from '@angular/core';
...
+  ) { }
+
+  dateValue: Date;
+
+  visible: boolean = false;
+
+  position: string = 'center';
+
+  showDialog(position: string) {
+      this.position = position;
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts:95]: can we hard code position in html, it's always going to be top right no?

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 6]
```diff
@@ -7,9 +18,210 @@ import { Component, OnInit } from '@angular/core';
...
+      this.visible = true;
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts:96]: name it better, isChildTableVisible

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 7]
```diff
@@ -7,9 +18,210 @@ import { Component, OnInit } from '@angular/core';
...
+      dateRange: [null],
+      start: [''],
+      end: ['']
+    });
+
+    this.exportLogForm.controls.searchOption.valueChanges.subscribe((searchTerm: string) => {
+      if (searchTerm) {
+        // This.expenseGroups.filter = searchTerm.trim().toLowerCase();
+      } else {
+        // This.expenseGroups.filter = '';
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts:195]: remove these if search is handled

File: src/app/integrations/si/main/export-log/export-log.component.ts (hunk 4 above)
Comment (ashwin1111) [reply src/app/integrations/si/main/export-log/export-log.component.ts:48]: can you resolve this / answer this?

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 8]
```diff
@@ -89,10 +89,7 @@ export class ExportLogComponent implements OnInit {
 
   visible: boolean = false;
 
-  position: string = 'center';
-
-  showDialog(position: string) {
-      this.position = position;
+  isChildTableVisible() {
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts]: showChildTable()

File: src/app/integrations/si/main/export-log/export-log.component.html [hunk 9]
```diff
@@ -60,7 +60,7 @@
                 </ng-template>
 
                 <ng-template pTemplate="body" let-rowData>
-                    <tr class="clickable-row" (click)="showDialog('top-right')">
+                    <tr class="clickable-row" (click)="isChildTableVisible()">
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.html]: showChildTable()

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 10]
```diff
@@ -70,6 +70,8 @@ export class ExportLogComponent implements OnInit {
 
   expenseGroups: ExpenseGroupList [];
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts:71]: name it filteredExpenseGroups and rename originalExpenseGroups to expenseGroups

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 11]
```diff
@@ -97,6 +99,25 @@ export class ExportLogComponent implements OnInit {
     window.open(url, '_blank');
   }
 
+  public filterTable(event: any) {
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts:110]: : Event

File: src/app/integrations/si/main/export-log/export-log.component.html [hunk 12]
```diff
@@ -117,27 +118,23 @@ <h4 class="tw-font-500 !tw-text-16-px tw-text-slightly-normal-text-color">No rec
...
-                    <td>{{rowData.expenses}}</td>
-                    <td>{{rowData.referenceNumber}}</td>
-                    <td>
-                        <div *ngIf="rowData.intacctUrl">
-                          <button (click)="openUrl(rowData.intacctUrl)" pTooltip="Open in New Tab" tooltipPosition="top">
-                            <svg-icon-sprite class="tw-pt-5-px" src="open-in-new-tab" width="20px" height="20px"></svg-icon-sprite>
-                          </button>
-                        </div>
-                    </td>
+                    <td>{{ rowData.expenses[0].expense_number }}</td>
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.html:134]: make it dynamic

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 13]
```diff
@@ -72,6 +72,8 @@ export class ExportLogComponent implements OnInit {
 
   originalExpenseGroups: ExpenseGroupList [];
 
+  childTableExpenseGroup: ExpenseGroupList [] = [];
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts:75]: call it expenses

File: src/app/integrations/si/main/export-log/export-log.component.ts [hunk 14]
```diff
@@ -93,8 +97,10 @@ export class ExportLogComponent implements OnInit {
 
   visible: boolean = false;
 
-  isChildTableVisible() {
-      this.visible = true;
+  isChildTableVisible(index: number) {
```
Comment (ashwin1111) [inline src/app/integrations/si/main/export-log/export-log.component.ts:100]: function shouldn't sound like boolean. `displayChildTable()`

File: src/app/integrations/si/main/export-log/export-log.component.ts (hunk 4 above)
Comment (anishfyle) [reply src/app/integrations/si/main/export-log/export-log.component.ts:48]: - apologies missed the comment, this is for the custom date range selector that we are allowing.
- the option which ideally would pop up the calendar.

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**